include/libtorrent/http_parser.hpp

```diff
@@ -60,14 +61,8 @@ namespace libtorrent {
 		enum flags_t { dont_parse_chunks = 1 };
 		explicit http_parser(int flags = 0);
 		~http_parser();
-		std::string const& header(char const* key) const
-		{
-			static std::string empty;
-			auto const i = m_header.find(key);
-			if (i == m_header.end()) return empty;
-			return i->second;
-		}
-
+		std::string const& header(string_view const key) const;
+		std::int64_t header_int(string_view const key, std::int64_t const def_value) const;
```

Comment:
there shouldn't be top-level ``const``s in declarations. Does your compiler suggest you do this?  They don't have any meaning in declarations and only add noise.